From 5ea7a94a662be06e74ba69f61282f9d5d7982af9 Mon Sep 17 00:00:00 2001
From: Peng Zhou <benjo.zhou@rock-chips.com>
Date: Tue, 7 Nov 2017 09:22:42 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3368-android: move cif from dtsi to
 dts

Move cif from rk3368-android.dtsi to rk3368-px5-evb-android.dts.
Redefine pinctrl for isp in rk3368-px5-evb-android.dts.

Change-Id: Ie8cd56cef507482b7d5e974a02f79279f0984aed
Signed-off-by: Peng Zhou <benjo.zhou@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3368-android.dtsi    |  4 ----
 arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi |  2 +-
 .../boot/dts/rockchip/rk3368-px5-evb-android.dts    | 13 +++++++++++++
 3 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
index d648abce8508..9f2acf90d268 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
@@ -260,10 +260,6 @@
 	status = "okay";
 };
 
-&rk3368_cif_sensor {
-	status = "okay";
-};
-
 &usb_otg {
 	status = "okay";
 	clocks = <&cru SCLK_OTGPHY0>, <&cru HCLK_OTG0>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi
index 33bde54a0431..bf0be36c0788 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi
@@ -1,7 +1,7 @@
 #include "../../../../../drivers/soc/rockchip/rk_camera_sensor_info.h"
 
 /{
-	rk3368_cif_sensor: rk3368_cif_sensor{
+	cif_sensor: cif_sensor {
 		compatible = "rockchip,sensor";
 		status = "okay";
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-px5-evb-android.dts b/arch/arm64/boot/dts/rockchip/rk3368-px5-evb-android.dts
index 3c18e1009a80..2e33ca3d6d7d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-px5-evb-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-px5-evb-android.dts
@@ -318,6 +318,10 @@
 	};
 };
 
+&cif_sensor {
+	status = "okay";
+};
+
 &cluster0_opp {
 	rockchip,threshold-freq = <408000>;
 	rockchip,freq-limit;
@@ -363,6 +367,15 @@
 	logic-supply = <&vdd_log>;
 };
 
+&isp {
+	pinctrl-names =
+		"isp_mipi_fl_prefl", "isp_flash_as_gpio",
+		"isp_flash_as_trigger_out";
+	pinctrl-0 = <&isp_prelight>;
+	pinctrl-1 = <&isp_flash_trigger_as_gpio>;
+	pinctrl-2 = <&isp_flash_trigger>;
+};
+
 &rockchip_suspend {
 	status = "okay";
 	rockchip,sleep-mode-config = <
-- 
2.35.3

